lib/tasks/scanning.rake

```diff
@@ -7,4 +9,47 @@ task :brakeman do
   end
 end
 
-task default: "brakeman"
+namespace :bundler do
+  desc 'Updates the ruby-advisory-db and runs audit'
+  task :audit do
+    %w(update check).each do |command|
+      Bundler::Audit::CLI.start [command]
+    end
+  end
+end
+
+namespace :yarn do
+  desc "Run yarn audit"
+  task :audit do
+    stdout, stderr, status = Open3.capture3("yarn audit --json")
+    unless status.success?
+      puts stderr
+      parsed = JSON.parse("[#{stdout.lines.join(",")}]")
+      puts JSON.pretty_generate(parsed)
+      if /503 Service Unavailable/.match?(stderr)
+        puts "Ignoring unavailable server"
+      elsif all_issues_ignored?(parsed)
+        puts "Ignoring known and accepted yarn audit results"
+      else
+        exit status.exitstatus
+      end
+    end
+  end
+end
+
+def all_issues_ignored?(issues)
+  summary = issues.find { |json| json["type"] == "auditSummary" }["data"]["vulnerabilities"]
+  # immediately fail if more findings are discovered, even if they have the same advisory ID,
+  # this helps to ensure that we fully evaluate the risk present
+  return false unless summary["moderate"] <= 5 && summary["high"] == 3 && summary["critical"] == 0
```

Comment:
[suggestion (nonblocking)] I think it'd be good to change `summary["high"] == 3` to `summary["high"] <= 3` so that we don't catch failures just because an advisory has been withdrawn. That being said, the exact check does encourage us to re-examine the results when anything changes, which isn't totally a bad thing.

Reply:
I think I'd like to spend a little time refactoring this next week so will make a small task to tackle next week so it doesn't block this PR.